src/App.jsx

```diff
@@ -134,6 +135,94 @@ function App() {
                 each year without running out of money for at least 30 years. This calculation 
```

Comment:
⚖️ **Midlevel Developer**

At my last company, we always separated modal components into their own files for better maintainability and to follow the Single Responsibility Principle (SRP). Right now, everything is in `App.jsx` (lines 135-94), which makes the file a bit bloated. I'd strongly recommend extracting the modal into a `<RuleModal />` component and maybe even creating a contexts folder for modal management if you plan to add more popups in the future. Also, can you add a dedicated test for opening/closing the modal and verifying the correct content loads?